base/sort.jl

```diff
@@ -245,6 +248,13 @@ abstract Algorithm
 immutable InsertionSortAlg <: Algorithm end
 immutable QuickSortAlg     <: Algorithm end
 immutable MergeSortAlg     <: Algorithm end
+immutable PartialQuickSort <: Algorithm
+    k::Int
+end
+
+# partially sort until the  end of the range
+PartialQuickSort(r::OrdinalRange) = PartialQuickSort(last(r))
```

Comment:
If we're going to include this, would it be good to support arbitrary ordinal ranges?